R/backend-sqlite.R

```diff
@@ -179,6 +179,12 @@ sql_query_join.SQLiteConnection <- function(con, x, y, vars, type = "inner", by
   }
 }
 
+#' @export
+values_prepare.SQLiteConnection <- function(con, df) {
```

Comment:
Is this also applicable with the new date+time extension via `extended_types = TRUE` ? Is there a way to find out if this has been passed to the connection?

https://rsqlite.r-dbi.org/reference/sqlite

Reply:
I didn't manage to create a date column without actually creating a new table...

``` r
con <- DBI::dbConnect(RSQLite::SQLite(), ":memory:", extended_types = TRUE)

# this does not produce columns of type DATE
out <- DBI::dbGetQuery(
  con,
  "SELECT
  date(`date`) AS `date`,
  CAST(`date` AS DATE) AS `date2`
FROM (
  SELECT
    NULL AS `date`
  WHERE (0 = 1)
  UNION ALL
  VALUES ('2020-01-01')
) AS `values_table`"
)

tibble::as_tibble(out)
#> # A tibble: 1 × 2
#>   date       date2
#>   <chr>      <int>
#> 1 2020-01-01  2020

# this works nicely
DBI::dbExecute(con, "create table tbl2 (aDate DATE)")
#> [1] 0
tibble::as_tibble(DBI::dbReadTable(con, "tbl2"))
#> # A tibble: 0 × 1
#> # … with 1 variable: aDate <date>
```

<sup>Created on 2022-05-30 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
Oh yeah, I'm not sure this will ever work. But then `copy_inline()` isn't necessarily faster or better on SQLite, so we could forward to `copy_to()`?

Reply:
Hmm, as far as I can see there are three reasons to use `copy_inline()`:

1. No permission to create a table -> SQLite does not have a concept of permissions apart from file permissions. So, this might not be relevant.
2. Performance -> probably not relevant for SQLite but one might test this.
3. Fully understandable queries in the sense that one can actually see in the query all the values used to create a table. This can sometimes be nice for tables with only a couple of rows.

The first two are probably irrelevant for SQLite. The third point can sometimes be nice.
But then it is a bit confusing to use `copy_inline()` and then not ending up with a `VALUES` clause. So, I'm not really sure this is a good idea.